lib/ingreedy/amount_parser.rb

```diff
@@ -59,7 +59,7 @@ class AmountParser < Parslet::Parser
     private
 
     def word_digits
-      Ingreedy.dictionaries.current.numbers.keys
+      Ingreedy.dictionaries.current.numbers.keys.sort_by(&:length).reverse
```

Comment:
Maybe we could also sort `numbers` when the current `Ingreedy::Dictionary` is [initialized](https://github.com/cookpad/ingreedy/blob/master/lib/ingreedy/dictionary.rb#L8)?